File: angular/packages/paperbase/src/lib/shared/tokens/paperbase-permissions.ts
export const PAPERBASE_PERMISSIONS = {
  Documents: {
    Default: 'Paperbase.Documents',
    Upload: 'Paperbase.Documents.Upload',
    Delete: 'Paperbase.Documents.Delete',
    PermanentDelete: 'Paperbase.Documents.PermanentDelete',
    Restore: 'Paperbase.Documents.Restore',
    Export: 'Paperbase.Documents.Export',
    ConfirmClassification: 'Paperbase.Documents.ConfirmClassification',
    Pipelines: {
      Default: 'Paperbase.Documents.Pipelines',
      Retry: 'Paperbase.Documents.Pipelines.Retry',
    },
    // Batch reprocessing of existing documents (#289) — admin-level.
    Reprocessing: {
      Default: 'Paperbase.Documents.Reprocessing',
      FieldExtraction: 'Paperbase.Documents.Reprocessing.FieldExtraction',
      Reclassification: 'Paperbase.Documents.Reprocessing.Reclassification',
    },
    Templates: {
      Default: 'Paperbase.Documents.Templates',
      Create: 'Paperbase.Documents.Templates.Create',
      Update: 'Paperbase.Documents.Templates.Update',
      Delete: 'Paperbase.Documents.Templates.Delete',
    },
  },
  Cabinets: {
    Default: 'Paperbase.Cabinets',
    Create: 'Paperbase.Cabinets.Create',
    Update: 'Paperbase.Cabinets.Update',
    Delete: 'Paperbase.Cabinets.Delete',
  },
  // Document-type schema management (#217) — admin-level, independent of document CRUD.
  DocumentTypes: {
    Default: 'Paperbase.DocumentTypes',
    Create: 'Paperbase.DocumentTypes.Create',
    Update: 'Paperbase.DocumentTypes.Update',
    Delete: 'Paperbase.DocumentTypes.Delete',
  },
  // Field-definition schema management (#217) — admin-level, independent of document CRUD.
  FieldDefinitions: {
    Default: 'Paperbase.FieldDefinitions',
    Create: 'Paperbase.FieldDefinitions.Create',
    Update: 'Paperbase.FieldDefinitions.Update',
    Delete: 'Paperbase.FieldDefinitions.Delete',
  },
} as const;

File: core/src/Dignite.Paperbase.Application.Contracts/Permissions/PaperbasePermissions.cs
using Volo.Abp.Reflection;

namespace Dignite.Paperbase.Permissions;

public class PaperbasePermissions
{
    public const string GroupName = "Paperbase";

    public static class Documents
    {
        public const string Default = GroupName + ".Documents";
        public const string Upload = Default + ".Upload";
        public const string Delete = Default + ".Delete";
        public const string PermanentDelete = Default + ".PermanentDelete";
        public const string Restore = Default + ".Restore";
        public const string Export = Default + ".Export";
        public const string ConfirmClassification = Default + ".ConfirmClassification";

        public static class Pipelines
        {
            public const string Default = Documents.Default + ".Pipelines";
            public const string Retry = Default + ".Retry";
        }
    }

    // 文件柜（#194）——人工组织维度，与 Documents 同级权限组。
    public static class Cabinets
    {
        public const string Default = GroupName + ".Cabinets";
        public const string Create = Default + ".Create";
        public const string Update = Default + ".Update";
        public const string Delete = Default + ".Delete";
    }

    public static string[] GetAll()
    {
        return ReflectionHelper.GetPublicConstantsRecursively(typeof(PaperbasePermissions));
    }
}
